src/main/java/com/segment/analytics/android/integrations/nielsendcr/NielsenDCRIntegration.java

```diff
@@ -33,6 +37,12 @@
   private Settings settings;
   int playheadPosition;
 
+  // reusable variables for `airdate` helper method
+  private final SimpleDateFormat formatter = new SimpleDateFormat("yyyyMMdd' 'HH:mm:ss");
```

Comment:
You should be able to make this `static` since it is the same across all instances of the class.

So `private final static SimpleDateFormat formatter`

That way it doesn't need to be instantiated for every class and can save memory if this class is instantiated multiple times.

Reply:
Ditto for next three lines. Generally constants can be `private static final`

Reply:
Brilliant!

Reply:
Are the `'` needed in the format?

Wouldn't this make a date that looks like:
`20190920' '14:10:11`

Reply:
(Once you make things `static final` convention then is to CASE_LIKE_THIS i.e. `FORMATTER` and `SHORT_DATE`)